src/common/snippets/src/lowered/pass/reduce_decomposition.cpp

```diff
@@ -60,7 +60,9 @@ bool ReduceDecomposition::run(LinearIR& linear_ir, LinearIR::constExprIt begin,
     for (auto expr_it = begin; expr_it != end; expr_it++) {
         const auto& reduce_expr = *expr_it;
         const auto& reduce = ov::as_type_ptr<ov::snippets::op::ReduceBase>(reduce_expr->get_node());
-        if (!reduce)
+        // Note: if Reduce can be casted to MemoryAccess, then it was already converted to TPP opset.
```

Comment:
Ok, removed TPP from the comment.

Callbacks are an interesting idea, but I'm not so sure about it because we don't have access to the transformations from the Subgraph node. The only solution I see so far is to pass an std::map<ov::DiscreteTypeInfo, std::function<bool(ExpressionPtr)>>

Reply:
> The only solution I see so far is to pass an std::map<ov::DiscreteTypeInfo, std::function<bool(ExpressionPtr)>>

This is exactly how ov::Model PassConfig works (please see [this file](https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/pass/pass_config.hpp#L163) for details). So I think we can implement the same approach in snippets manager's pass config (within a separate activity)